lib/etl.ex

```diff
@@ -87,59 +79,96 @@ defmodule Etl do
     |> Enum.each(fn {{mod, ref}, pass, fail} -> mod.ack(ref, pass, fail) end)
   end
 
-  defp group_by_status({key, messages}) do
-    {pass, fail} =
-      Enum.reduce(messages, {[], []}, fn
-        %{status: :ok} = msg, {pass, fail} ->
-          {[msg | pass], fail}
+  defp start_steps(graph, [], _context), do: graph
 
-        msg, {pass, fail} ->
-          {pass, [msg | fail]}
-      end)
+  defp start_steps(graph, [step | remaining], context) do
+    starter = &start_step(&1, context, remaining == [])
```

Comment:
not sure how i feel about this one since i don't know that i've ever seen a capture syntax used to bind a function to a variable; only when a function is called. why is this bound to a variable and called as a lambda instead of just called `start_step/3` in the body of the case statement below?